Assets/Plugins/Source/Editor/Utility/GUIEditorUtility.cs

```diff
@@ -154,46 +152,43 @@ public static void AssigningBoolField(string label, ref bool value, float labelW
             EditorGUIUtility.labelWidth = originalLabelWidth;
         }
 
-        public static void AssigningFloatToStringField(string label, ref string value, float labelWidth = -1, string tooltip = null)
+        public static void AssigningFloatToStringField(string label, ref float? value, float labelWidth = -1, string tooltip = null)
```

Comment:
I'm looking at this and it *feels* like this is so similar to the other function, it's a sign we want to make things more generic somehow. But I can't really think of a way to do that in a way that's actually cleaner or better than just repeating ourselves...

> delegate bool TryParseFunc<T>(string input, out T output);
> public static void AssigningValueToStringField<T>(string label, ref T? value,  TryParseFunc<T> tryParseFunc, float labelWidth = -1, string tooltip = null)
> ...

Is that any better? Maybe we have this wrapper function signature that calls an inner function? Iunno, spitballing. I'd be fine with this as is.

Reply:
Because eventually these will all use the `RenderInput` methods that inspect the config fields and do all the rendering, I opted not to optimize these functions since they will be going away in the near future.